pkg/web/web.go

```diff
@@ -0,0 +1,94 @@
+package web
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/gorilla/mux"
+	"go.signoz.io/signoz/pkg/http/middleware"
+	"go.uber.org/zap"
+)
+
+var _ http.Handler = (*Web)(nil)
+
+const (
+	indexFileName string = "index.html"
+)
+
+type Web struct {
+	logger *zap.Logger
+	cfg    Config
+}
+
+func New(logger *zap.Logger, cfg Config) (*Web, error) {
+	if logger == nil {
+		return nil, fmt.Errorf("cannot build web, logger is required")
+	}
+
+	fi, err := os.Stat(cfg.Directory)
+	if err != nil {
+		return nil, fmt.Errorf("cannot access web directory: %w", err)
+	}
+
+	ok := fi.IsDir()
+	if !ok {
+		return nil, fmt.Errorf("web directory is not a directory")
+	}
+
+	fi, err = os.Stat(filepath.Join(cfg.Directory, indexFileName))
+	if err != nil {
+		return nil, fmt.Errorf("cannot access %q in web directory: %w", indexFileName, err)
+	}
+
+	if os.IsNotExist(err) || fi.IsDir() {
+		return nil, fmt.Errorf("%q does not exist", indexFileName)
+	}
+
+	return &Web{
+		logger: logger.Named("go.signoz.io/pkg/web"),
+		cfg:    cfg,
+	}, nil
+}
+
+func (web *Web) AddToRouter(router *mux.Router) error {
+	cache := middleware.NewCache(7 * 24 * time.Hour)
```

Comment:
The `NewCache` reads as if we are maintaining the cache rather than the middleware that controls the cache headers. Can every web asset be cached?

Reply:
Yeah every web asset can be cached. If we were doing SSR, it would not have been possible. A static build does not change once built.

I hope I understood the question correctly!